structuredheadings/plugin.js

```diff
@@ -345,69 +347,20 @@
           this.add("restart", "Restart Styling", "Restart Styling");
         },
 
-        clearAutonumberClassesForHeading: function (heading) {
-          clearStyles(editor, heading);
-          cssUtils.clearLevel(editor, heading);
-          cssUtils.clearNumbering(editor, heading);
-        },
-
-        setAutonumberClassesForHeading: function (value, heading) {
-          cssUtils.setNumbering(editor, heading);
-          cssUtils.setLevel(editor, heading);
-          setCurrentStyle(editor, heading, value);
-        },
-        onClick: function (value) {  // eslint-disable-line max-statements
+        onClick: function (value) {
           editor.fire("saveSnapshot");
 
           if (isInList(editor, editor.elementPath())) {
             editor.execCommand("applyPresetToList", value);
-            return;
-          }
-
-          if (value === "restart") {
+          } else if (value === "restart") {
             editor.execCommand("restartNumbering");
-            editor.fire("saveSnapshot");
-            return;
-          }
-          var selection = editor.getSelection();
-          var headings = CKEDITOR.plugins.structuredheadings.getHeadingsInSelection(
-                editor,
-                selection
-              );
-          var func;
-
-          if (value === "clear") {
-            func = this.clearAutonumberClassesForHeading;
           } else {
-            func = this.setAutonumberClassesForHeading.bind(this, value);
-
-             // prep the html for the collapsed, not in a heading case
-            if (headings === null) {
-              // avoid an extra snapshot created by the matchHeading command
-              editor.fire("lockSnapshot", { dontUpdate: true });
-              editor.execCommand("matchHeading");
-              editor.fire("unlockSnapshot");
-              // put the new heading into the headings array
-              headings = [CKEDITOR.plugins.structuredheadings.getCurrentBlockFromPath(editor)];
-            }
```

Comment:
much of this stuff was moved out into an actual command object.  it was all a bit intertwined